### PR TITLE
DB: Update queries and db structure to lastest TrinityCore cha…

### PR DESCRIPTION
**Pull request by Carbenium**
…nges (TrinityCore/TrinityCore@6b43e72f019b6a91be812f4e4da58a4dae7c21a6)

### CODE REVIEW

**Comment by Sarjuuk** [comment]
Thanks for the heads up.
I'm currently combing through the changes as there are some changes, that are unnecessary or affect code, that wasn't also updated.

I'll edit them in later.

File: pages/skill.php [hunk 1]
```diff
@@ -278,7 +278,7 @@ protected function generateContent()
             {
                 $this->addJS('?data=zones&locale='.User::$localeId.'&t='.$_SESSION['dataKey']);
 
-                $trainer = new CreatureList(array(CFG_SQL_LIMIT_NONE, ['ct.id', $list], ['s.guid', NULL, '!'], ['ct.npcflag', 0x10, '&']));
+                $trainer = new CreatureList(array(CFG_SQL_LIMIT_NONE, ['ct.ID', $list], ['s.guid', NULL, '!'], ['ct.npcflag', 0x10, '&']));
```
Comment (Sarjuuk) [inline pages/skill.php]: aowow_creature did not change. 

zealous find 'n replace?

File: pages/spell.php [hunk 2]
```diff
@@ -1032,7 +1032,7 @@ protected function generateContent()
 
             if ($list)
             {
-                $tbTrainer = new CreatureList(array(CFG_SQL_LIMIT_NONE, ['ct.id', $list], ['s.guid', NULL, '!'], ['ct.npcflag', 0x10, '&']));
+                $tbTrainer = new CreatureList(array(CFG_SQL_LIMIT_NONE, ['ct.ID', $list], ['s.guid', NULL, '!'], ['ct.npcflag', 0x10, '&']));
```
Comment (Sarjuuk) [inline pages/spell.php]: same as in pages/skill.php?

File: setup/tools/sqlgen/source.func.php [hunk 3]
```diff
@@ -1045,9 +1045,9 @@ function taughtSpell($item)
     CLISetup::log('   * #4  Quest');
     $quests = DB::World()->select('
         SELECT spell AS ARRAY_KEY, id, SUM(qty) AS qty, BIT_OR(side) AS side FROM (
```
Comment (Sarjuuk) [inline setup/tools/sqlgen/source.func.php]: you'll probably want to also change `id` to uppercase here and consequently in line 1065.
But i'm not sure how strict mysql goes about that.

File: pages/skill.php (hunk 1 above)
Comment (Carbenium) [inline pages/skill.php]: Oh yeah, right. That got mixed up. I thoght on the first sight it was referencing npc_trainer...

**Comment by Sarjuuk** [comment]
https://github.com/TrinityCore/TrinityCore/pull/14941
bloody hell!  
better to wait this one out and add both at once

**Comment by Carbenium** [comment]
Yeah. You're right. I totally forgot about that one. It should be merged soon.

**Comment by Carbenium** [comment]
TC merge done https://github.com/TrinityCore/TrinityCore/commit/adb5d3ea5f1acc44b3799b08297343ad1413bc8d
Let's start the work :D

**Comment by Sarjuuk** [comment]
The impact is actually only minor. I didn't even use these fields directly, although they are copied over.
They are just used on setup to categorize more GOs as quest-related.

I think it's best to drop them altogether.

**Comment by Carbenium** [comment]
More changes coming: https://github.com/TrinityCore/TrinityCore/pull/14982

**Comment by ShinDarth** [comment]
You are going to hate me, aren't you? :D

**Comment by Sarjuuk** [comment]
i already do, i already do...

;P

**Comment by Sarjuuk** [comment]
@ShinDarth  are .. you done 

*careful optimism*

**Comment by ShinDarth** [comment]
I don't know, maybe other changes can be backported. I need to discuss with the others:

https://gist.github.com/ShinDarth/20a32dd47cfbc37cab43

**Comment by Carbenium** [comment]
@Sarjuuk : how is it going with the changes?

**Comment by Sarjuuk** [comment]
Have yet to touch it again. 
Probably on Sunday.

**Comment by Sarjuuk** [comment]
well ... it *is* sunday!

**Pull request closed without merging**